shared/utility/support/evil/pure_evil.cpp

```diff
@@ -21,30 +40,31 @@ extern "C" {
 void error_callback(void* /*data*/, const char* /*msg*/, int /*errnum*/) {}
 
 // Initialise our state for the backtrace (I think we can do this once per binary)
-// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables,cert-err58-cpp)
-backtrace_state* state = backtrace_create_state(nullptr, int(true), error_callback, nullptr);
+// Ignore the fact this can potentially throw an exception
+// NOLINTNEXTLINE(cert-err58-cpp,cppcoreguidelines-avoid-non-const-global-variables)
+backtrace_state* state = backtrace_create_state(nullptr, 1, error_callback, nullptr);
 
-using cxa_throw_func_type = void (*)(void*, void*, void (*)(void*));
+// We can't use a using statement here as the attribute on the end will make it explode
+// NOLINTNEXTLINE(modernize-use-using,cppcoreguidelines-avoid-non-const-global-variables)
+typedef void (*cxa_throw_func_type)(void*, void*, void (*)(void*)) __attribute__((noreturn));
```

Comment:
What's the benefit of having the attribute?

Reply:
Because it has to match the signature of the cxa throw function which has this attribute